src/CSET/operators/read.py

```diff
@@ -423,6 +426,66 @@ def _fix_pressure_coord_name_callback(cube: iris.cube.Cube):
             coord.rename("pressure")
 
 
+def _lfric_time_callback(cube: iris.cube.Cube):
+    """
+    Fix time coordinate metadata if missing dimensions.
+
+    Some model data does not contain forecast_reference_time or forecast_period as
+    expected coordinates, and so we cannot aggregate over case studies without this
+    metadata. This callback fixes these issues.
+
+    Notes
+    -----
+    Some parts of the code have been adapted from Paul Earnshaw's scripts.
+    """
+    # Construct forecast_reference time and forecast_period if they dont exist.
+    if not cube.coords("forecast_reference_time"):
+        try:
+            tcoord = cube.coord("time")
+            init_time = datetime.datetime.fromisoformat(
+                cube.coord("time").attributes["time_origin"]
+            )
+            time_unit = cube.coord("time").units
+            frt_point = cf_units.date2num(
+                init_time, str(time_unit), calendar=time_unit.calendar
+            )
```

Comment:
```suggestion
            frt_point = time_unit.date2num(init_time)
```
You can directly use the method on the units object to do this a bit more cleanly.